frontend/packages/shared/src/queryInvalidator/SyncSuccessQueriesInvalidator.ts

```diff
@@ -15,22 +15,33 @@ import { Queue } from 'app-shared/queue/Queue';
 
 export class SyncSuccessQueriesInvalidator extends Queue {
   private static instance: SyncSuccessQueriesInvalidator | null = null;
-  private org: string;
-  private app: string;
-  private queryClient: QueryClient;
+  private _org: string;
+  private _app: string;
+  private _layoutSetName?: string;
+  private _queryClient: QueryClient;
 
   // Maps file names to their cache keys for invalidation upon sync success - can be extended to include more files
   private readonly fileNameCacheKeyMap: Record<string, Array<QueryKey | string>> = {
     'applicationmetadata.json': [QueryKey.AppMetadata, '[org]', '[app]'],
     'layout-sets.json': [QueryKey.LayoutSets, '[org]', '[app]'],
     'policy.xml': [QueryKey.AppPolicy, '[org]', '[app]'],
+    'Settings.json': [QueryKey.FormLayoutSettings, '[org]', '[app]', '[layoutSetName]'],
   };
 
+  // Maps folder names to their cache keys for invalidation upon sync success - can be extended to include more folders
+  private readonly folderNameCacheKeyMap: Record<string, Array<QueryKey | string>> = {
+    layouts: [QueryKey.FormLayouts, '[org]', '[app]', '[layoutSetName]'],
+  };
+
+  public setLayoutSetName(layoutSetName: string): void {
+    this._layoutSetName = layoutSetName;
+  }
```

Comment:
Ref. previous comment, using a setter makes the code more readable because it is a property update within the class. 

_Note: Setters do not have a return value_

```suggestion
  public set layoutSetName(layoutSetName: string) {
    this._layoutSetName = layoutSetName;
  }
```